fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1215,6 +1217,72 @@ void continuationDuringRepartitioningTest(boolean isGrouped,
+
+        final RecordLayerPropertyStorage contextProps = RecordLayerPropertyStorage.newBuilder()
+                .addProp(LuceneRecordContextProperties.LUCENE_REPARTITION_DOCUMENT_COUNT, 6)
+                .addProp(LuceneRecordContextProperties.LUCENE_REPARTITION_MAX_DOCUMENT_COUNT, 10)
+                .build();
+
+        final int group = 1;
+        final Tuple groupTuple = Tuple.from(group);
+        final long start = Instant.now().toEpochMilli();
+        final String luceneSearch = "text:about";
+
+        final int docCount = 25;
+        List<Tuple> primaryKeys = new ArrayList<>();
+        try (FDBRecordContext context = openContext(contextProps)) {
+            schemaSetup.accept(context);
+            for (int i = 0; i < docCount; i++) {
+                ComplexDocument cd = ComplexDocument.newBuilder()
+                        .setGroup(group)
+                        .setDocId(1000L + i)
+                        .setIsSeen(true)
+                        .setText("A word about what I want to say")
+                        .setTimestamp(start + i * 100)
+                        .setHeader(ComplexDocument.Header.newBuilder().setHeaderId(1000L - i))
+                        .build();
+                final Tuple primaryKey;
+                primaryKey = recordStore.saveRecord(cd).getPrimaryKey();
+                primaryKeys.add(primaryKey);
+            }
+            commit(context);
+        }
+
+        // initially, all documents are saved into one partition
+        List<LucenePartitionInfoProto.LucenePartitionInfo> partitionInfos = getPartitionMeta(index,
+                groupTuple, contextProps, schemaSetup);
+        assertEquals(1, partitionInfos.size());
+        assertEquals(docCount, partitionInfos.get(0).getCount());
+
+        // run re-partitioning
+        explicitMergeIndex(index, contextProps, schemaSetup);
+
+        // even though we capped the documents moved on every partition to 10, we still
+        // repartitioned as expected
+        //  partition 0: with docs 18 - 24
+        //  partition 3: with docs 12 - 17
+        //  partition 2: with docs  6 - 11
+        //  partition 1: with docs  0 - 5
+        try (FDBRecordContext context = openContext(contextProps)) {
+            schemaSetup.accept(context);
+            // due to cap, the rebalance should have been called 4 times
+            assertEquals(4, getCounter(context, LuceneEvents.Counts.LUCENE_REBALANCE_CALLS).getCount());
+
+            validateDocsInPartition(index, 0, groupTuple, Set.copyOf(primaryKeys.subList(18, 25)), luceneSearch);
+            validateDocsInPartition(index, 3, groupTuple, Set.copyOf(primaryKeys.subList(12, 18)), luceneSearch);
+            validateDocsInPartition(index, 2, groupTuple, Set.copyOf(primaryKeys.subList(6, 12)), luceneSearch);
+            validateDocsInPartition(index, 1, groupTuple, Set.copyOf(primaryKeys.subList(0, 6)), luceneSearch);
+        }
```

Comment:
Nice test!
Can it be repeated with different variety of numbers? Something like: 
```
@Test 
void capDocCountMovedDuringRepartitioningTestA() throws IOException {
  capDocCountMovedDuringRepartitioningTest(/*docCount*/ 73, /*highWatermark*/ 23, /*transactionDocCount*/ 7);
}

@Test 
void capDocCountMovedDuringRepartitioningTestA() throws IOException {
  capDocCountMovedDuringRepartitioningTest(/*docCount*/ 17, /*highWatermark*/ 10, /*transactionDocCount*/ 3);
}

Etc. 
```